src/node.h

```diff
@@ -148,6 +148,77 @@ typedef intptr_t ssize_t;
 
 namespace node {
 
+  enum NodeInstanceType { MAIN, WORKER };
+
+  class NodeInstanceData {
+    public:
+
+
+      NodeInstanceData(NodeInstanceType node_instance_type,
+                       uv_loop_t* loop,
+                       int argc,
+                       const char** argv,
+                       int exec_argc,
+                       const char** exec_argv) :
+          node_instance_type_(node_instance_type),
+          loop_(loop),
+          argc_(argc),
+          argv_(argv),
+          exec_argc_(exec_argc),
+          exec_argv_(exec_argv) {
+      }
+
+      inline uv_loop_t* loop() {
+        assert(loop_);
+        return loop_;
+      }
+
+      inline int exit_code() {
+        assert(is_main());
+        return exit_code_;
+      }
+
+      inline void set_exit_code(int exit_code) {
+        assert(is_main());
+        exit_code_ = exit_code;
+      }
+
+      inline bool is_main() {
+        return node_instance_type_ == MAIN;
+      }
+
+      inline bool is_worker() {
+        return node_instance_type_ == WORKER;
+      }
+
+      inline int argc() {
+        return argc_;
+      }
+
+      inline const char** argv() {
+        return argv_;
+      }
+
+      inline int exec_argc() {
+        return exec_argc_;
+      }
+
+      inline const char** exec_argv() {
+        return exec_argv_;
+      }
```

Comment:
If you're defining the functions inline, you don't have to mark them `inline`.  That's only for when you break out the definition from the declaration.